File: serve_web.py
#!/usr/bin/env python3
"""
Simple HTTP server for Century Old Tunes web frontend
Run this to serve the docs/ folder locally
"""

import http.server
import socketserver
import os
import sys

PORT = 8000
DIRECTORY = "docs"

class Handler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)

if __name__ == "__main__":
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    
    with socketserver.TCPServer(("", PORT), Handler) as httpd:
        print(f"🎵 Century Old Tunes Web Server")
        print(f"📁 Serving {DIRECTORY}/ directory")
        print(f"🌐 Open: http://localhost:{PORT}")
        print(f"⏹️  Press Ctrl+C to stop\n")
        
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\n👋 Server stopped")
            sys.exit(0)

File: R/serve_web.R
# Serve the web frontend using R's built-in server
# Alternative to Python server

serve_web_frontend <- function(port = 8000) {
  if (!requireNamespace("servr", quietly = TRUE)) {
    cat("Installing servr package for web server...\n")
    install.packages("servr")
  }
  
  library(servr)
  
  cat("🎵 Century Old Tunes Web Server\n")
  cat("📁 Serving docs/ directory\n")
  cat("🌐 Open: http://localhost:", port, "\n")
  cat("⏹️  Press Ctrl+C or ESC to stop\n\n")
  
  # Change to project root and serve docs folder
  servr::httd(dir = "docs", port = port, browser = TRUE)
}

# Run the server
serve_web_frontend()